src/loader.ts

```diff
@@ -241,6 +136,53 @@ interface ModuleDefinitionArguments extends Array<any> {
 		return has;
 	})();
 
+	let listenerQueues = {};
+
+	const reportModuleLoadError = function (parent: Module, module: Module, url: string): void {
+		let parentMid = (parent ? ' (parent: ' + parent.mid + ')' : '');
+		let message = `Failed to load module ${module.mid} from ${url}${parentMid}`;
+		let error = mix<LoaderError>(new Error(message), {
+			src: 'dojo/loader',
+			info: {
+				module,
+				url,
+				parentMid
+			}
+		});
+
+		if (!emit('error', error)) { throw error; };
+	};
+
+	const emit = function(type: SignalType, args: {}): number | boolean {
+		let queue: any[] = listenerQueues[type];
+		let hasListeners = queue && queue.length;
+
+		if (hasListeners) {
+			for (let listener of queue.slice(0)) {
+				listener.apply(null, Array.isArray(args) ? args : [args]);
+			}
+		}
+
+		return hasListeners;
+	};
+
+	const on = function(type: string, listener: (error: LoaderError) => any): { remove: () => void } {
+		let queue = listenerQueues[type] || (listenerQueues[type] = []);
+
+		queue.push(listener);
+
+		return {
+			remove() : void {
+				for (let i = 0; i < queue.length; i++) {
```

Comment:
Yeah, but you would have to guard the `indexOf` I think.  So doing something like:

```typescript
const idx = queue.indexOf(listener);
if (idx >= 0) {
    queue.splice(idx, 1);
}
```

I don't know which is faster, maybe a JSPerf is in order?  Likely immaterial, but an `indexOf` just seems cleaner to me.  Which do you prefer?